.github/workflows/pr.yml

```diff
@@ -0,0 +1,34 @@
+name: Pull Request Checks
+
+on: [pull_request]
+
+jobs:
+
+  changelog-check:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: Check Changelog Updated
+        run: |
+          set -Eeuo pipefail
+          log() { echo "==> $*" 1>&2; }
+          die() { log "FATAL: $*"; return 1; }
+
+          VERSION="$(cat release/VERSION)"
+          NOTES_FILE="release/changes/$VERSION.md"
+          [ -f "$NOTES_FILE" ] || {
+            die "No release notes found at $NOTES_FILE"
+          }
+
+          PR_NUMBER="$(basename "${GITHUB_REF%/merge}")"
+
+          FILES_URL="/repos/${{ github.repository}}/pulls/$PR_NUMBER/files"
+          CHANGED_FILES_RESPONSE="$(
+            curl \
+              -H "Accept: application/vnd.github.v3+json" \
+              -H 'Authorization: Bearer ${{ secrets.GITHUB_TOKEN }}' \
+              "$FILES_URL"
+          )"
+
+          echo "$CHANGED_FILES_RESPONSE"
```

Comment:
Could there be an advantage if putting this into its own script, and using this step to just call it? I see we do that for some functionality under ``/scripts`` already.

Reply:
Yes, definitely! I'd forgotten about this PR tbh, it should be "draft" as it was only half done iirc.

Reply:
I'll come back to this soon.